src/main/scala/vsys/db/SubStorage.scala

```diff
@@ -0,0 +1,27 @@
+package vsys.db
+
+import com.google.common.primitives.{Bytes, Ints}
+import org.iq80.leveldb.{DB, WriteBatch}
+
+class SubStorage(db: DB, name: String) extends Storage(db) {
+
+  private val subPrefix: Array[Byte] = name.getBytes(Charset)
+
+  override protected def makePrefix(prefix: Array[Byte]): Array[Byte] = Bytes.concat(subPrefix, Separator, prefix, Separator)
+
+  override protected def makeKey(prefix: Array[Byte], key: Array[Byte]): Array[Byte] = Bytes.concat(subPrefix, Separator, prefix, Separator, key)
+
+  override protected def makeKey(prefix: Array[Byte], key: String): Array[Byte] = makeKey(prefix, key.getBytes(Charset))
+
+  override protected def makeKey(prefix: Array[Byte], key: Int): Array[Byte] = makeKey(prefix, Ints.toByteArray(key))
+
+  override def removeEverything(b: Option[WriteBatch]): Unit = {
+    val it = allKeys
+    while (it.hasNext) {
+      val key = it.next()
+      if (key.startsWith(subPrefix)) delete(key, b)
```

Comment:
does the iterator still work after the key is deleted?

Reply:
Yes, it still works. 

> The result of DB::NewIterator() presents a stable view of the 
database contents as of the time the iterator was created. 
I.e., you can delete or make other changes to the database 
after this call; those will have no effect on the state returned 
by the iterator.  
> In other words, an iterator provides a light-weight snapshot. 
> This should be documented better. 

Quoted from [google leveldb group]( https://groups.google.com/forum/#!topic/leveldb/QZS1F3lTg4s)